tests/resolution/check/test_check_deprecated_addon.py

```diff
@@ -0,0 +1,75 @@
+"""Test check for deprecated addons."""
+
+from unittest.mock import patch
+
+from supervisor.addons.addon import Addon
+from supervisor.const import AddonStage, CoreState
+from supervisor.coresys import CoreSys
+from supervisor.resolution.checks.deprecated_addon import CheckDeprecatedAddon
+from supervisor.resolution.const import ContextType, IssueType
+
+
+async def test_base(coresys: CoreSys):
+    """Test check basics."""
+    deprecated_addon = CheckDeprecatedAddon(coresys)
+    assert deprecated_addon.slug == "deprecated_addon"
+    assert deprecated_addon.enabled
+
+
+async def test_check(coresys: CoreSys, install_addon_ssh: Addon):
+    """Test check for deprecated addons."""
+    deprecated_addon = CheckDeprecatedAddon(coresys)
+    await coresys.core.set_state(CoreState.SETUP)
+
+    await deprecated_addon()
+    assert len(coresys.resolution.issues) == 0
+
+    # Mock test addon as deprecated
+    install_addon_ssh.data["stage"] = AddonStage.DEPRECATED
+
+    await deprecated_addon()
+
+    assert len(coresys.resolution.issues) == 1
+    assert coresys.resolution.issues[0].type is IssueType.DEPRECATED_ADDON
+    assert coresys.resolution.issues[0].context is ContextType.ADDON
+    assert coresys.resolution.issues[0].reference == install_addon_ssh.slug
+    assert len(coresys.resolution.suggestions) == 0
```

Comment:
```suggestion
    assert len(coresys.resolution.suggestions) == 1
```
Or adjust to assert for the specific suggestion that we're now adding to fix the test.